services/api-server/src/simcore_service_api_server/api/dependencies/rabbitmq.py

```diff
@@ -18,6 +23,17 @@ def get_log_distributor(app: Annotated[FastAPI, Depends(get_app)]) -> LogDistrib
     return cast(LogDistributor, app.state.log_distributor)
 
 
+async def wait_till_log_distributor_ready(app) -> None:
+    start = time()
+    while not hasattr(app.state, "log_distributor"):
+        if time() - start > _MAX_WAIT_FOR_LOG_DISTRIBUTOR_SECONDS:
+            raise ApplicationSetupError(
+                f"Api server's log_distributor was not ready within {_MAX_WAIT_FOR_LOG_DISTRIBUTOR_SECONDS=} seconds"
+            )
+        await asyncio.sleep(1)
```

Comment:
could you please do this with tenacity? We're trying to remove all the while style waiting.

Reply:
done, thanks